src/reanimated2/helperTypes.ts

```diff
@@ -118,22 +97,19 @@ type SharedTransitionProps = {
   sharedTransitionStyle?: SharedTransition;
 };
 
-type AnimatedPropsProp<P extends object> = NonStyleAnimatedProps<P> &
-  JustStyleAnimatedProp<P> &
-  StyleAnimatedProps<P> &
+type AnimatedPropsProp<Props extends object> = NonStyleAnimatedProps<Props> &
```

Comment:
I am afraid that `AnimatedPropsProp` may be a confusing name

Reply:
It is confusing, I will change it once time comes for `AnimatedProps`.